arkouda/segarray.py

```diff
@@ -588,12 +594,20 @@ def remove_repeats(self, return_multiplicity=False):
         """
         isrepeat = zeros(self.values.size, dtype=akbool)
         isrepeat[1:] = self.values[:-1] == self.values[1:]
-        isrepeat[self.segments] = False
+        isrepeat[self.segments[self._non_empty]] = False
         truepaths = self.values[~isrepeat]
         nhops = self.grouping.sum(~isrepeat)[1]
-        # truehops = ak.cumsum(~isrepeat)
-        # nhops = ak.concatenate((truehops[self.segments[1:]], ak.array([truehops.sum()+1]))) - truehops[self.segments]
         truesegs = cumsum(nhops) - nhops
+        # Correct segments to properly assign empty lists - prevents dropping empty segments
+        if not self._non_empty.all():
+            empty_segs = []
+            for i in range(self.size):
+                if not self._non_empty[i]:
+                    if i < truesegs.size:
+                        empty_segs.append(truesegs[i])
+                    else:
+                        empty_segs.append(truepaths.size)
+            truesegs = concatenate([truesegs, array(empty_segs)])
```

Comment:
I think this simplifies this a bit, it's still not ideal (i dont love the way `trailing_empties` is constructed). I tried to use `ak.where` but the lengths didn't line up the way i wanted
```python
if not self._non_empty.all():
    # empties after the first truesegs.size will be out of bounds, we append truepaths.size to compensate
    trailing_empties = array([truepaths.size] * (self.size - truesegs.size))
    empty_segs = concatenate([truesegs, trailing_empties])[~self._non_empty]
    truesegs = concatenate([truesegs, empty_segs])
```
also just for fun.... awful one liner! lol
```python
truesegs = truesegs if self._non_empty.all() else concatenate([truesegs, concatenate([truesegs, array([truepaths.size] * (self.size - truesegs.size))])[~self._non_empty]])

```

Reply:
note: we might want to update this once `ak.full` from #1207 is implemented

Reply:
Updated this to avoid using the loop.